### PR TITLE
S3Client improvements

### PR DESCRIPTION
**Pull request by adilkhash**
## Description
Fixes hardcoded UTF-8 encoding inside `S3Client.get_as_string` method

## Motivation and Context
Issue: https://github.com/spotify/luigi/issues/2568

## Have you tested this? If so, how?
Yes. Tested running `tox`

### CODE REVIEW

File: luigi/contrib/s3.py [hunk 1]
```diff
@@ -388,17 +388,19 @@ def get(self, s3_path, destination_local_path):
...
         """
         (bucket, key) = self._path_to_bucket_and_key(s3_path)
-
-        # get the content
         obj = self.s3.Object(bucket, key)
-        contents = obj.get()['Body'].read().decode('utf-8')
+        contents = obj.get()['Body'].read()
         return contents
 
+    def get_as_string(self, s3_path, encoding='utf-8'):
```
Comment (dlstadther) [inline luigi/contrib/s3.py]: Could you add a docstring for the new `get_as_string()`?

File: test/contrib/s3_test.py [hunk 2]
```diff
@@ -314,6 +314,15 @@ def test_get(self):
...
 
+    def test_get_as_bytes(self):
+        create_bucket()
+        s3_client = S3Client(AWS_ACCESS_KEY, AWS_SECRET_KEY)
+        s3_client.put(self.tempFilePath, 's3://mybucket/putMe')
+
+        contents = s3_client.get_as_bytes('s3://mybucket/putMe')
+
+        self.assertEquals(contents, self.tempFileContents)
+
```
Comment (dlstadther) [inline test/contrib/s3_test.py]: Could you also add 2 other get_as_string tests? Maybe one for utf-8 and one for another encoding?

File: luigi/contrib/s3.py (hunk 1 above)
Comment (adilkhash) [reply luigi/contrib/s3.py]: Done

File: test/contrib/s3_test.py (hunk 2 above)
Comment (adilkhash) [reply test/contrib/s3_test.py]: Done

**Comment by adilkhash** [comment]
Do you know why tests on `nonhdfs` fail?

File: luigi/contrib/s3.py [hunk 3]
```diff
@@ -388,17 +388,22 @@ def get(self, s3_path, destination_local_path):
...
+        """
+        Get the contents of an object stored in S3 as string.
```
Comment (dlstadther) [inline luigi/contrib/s3.py]: sorry to be nit-picky, but could you add the param docstring descriptions too?

**Comment by dlstadther** [comment]
I'm not yet sure of the Travis failures. I'm seeing the same errors on all recent PRs. I'll investigate as time allows

File: luigi/contrib/s3.py (hunk 3 above)
Comment (adilkhash) [reply luigi/contrib/s3.py]: Done

**Comment by dlstadther** [comment]
Could you merge the master test fix into this branch?

**Comment by adilkhash** [comment]
Done

**Pull request merged**